src/HHbbVV/processors/bbVVSkimmer.py

```diff
@@ -684,6 +692,373 @@ def process(self, events: ak.Array):
+                E = np.sqrt(m**2 + px**2 + py**2 + pz**2)
+                return E, px, py, pz
+
+            def compute_spatial_part(E, px, py, pz, boost_vector, boost_vector_dot_product):
+                gamma = 1.0 / np.sqrt(1.0 - boost_vector_dot_product)
+
+                # Dot product of momentum and boost vector
+                p_dot_v = px * boost_vector[0] + py * boost_vector[1] + pz * boost_vector[2]
+
+                # Lorentz transformation for the spatial components
+                px_prime = (
+                    px
+                    + (gamma - 1) * p_dot_v * boost_vector[0] / boost_vector_dot_product
+                    - gamma * E * boost_vector[0]
+                )
+                py_prime = (
+                    py
+                    + (gamma - 1) * p_dot_v * boost_vector[1] / boost_vector_dot_product
+                    - gamma * E * boost_vector[1]
+                )
+                pz_prime = (
+                    pz
+                    + (gamma - 1) * p_dot_v * boost_vector[2] / boost_vector_dot_product
+                    - gamma * E * boost_vector[2]
+                )
+
+                return px_prime, py_prime, pz_prime
+
+            # seperation between both ak8 higgs jets
+            # print(f"\nTime taken defining: {time.time()-start_time:.6f} seconds")
+
+            vbfVars[f"vbf_dR_HH"] = VVJet.deltaR(
+                bbJet
+            )  # may have to treat same as nGoodVBFJets. same holds for other 1d arrays
+            # print(f"\nTime after inserting one item: {time.time()-start_time:.6f} seconds")
+
+            # ∆R distance between H-VV and the leading VBF-jet:
+            # vbfVars[f"vbf_dR_HVV"] = pad_val(VVJet.delta_r(vbfJets_sorted_pt), 2, axis=1)
+
+            vbfVars[f"vbf_dR_j0_HVV"] = vbf1.deltaR(VVJet)
+            vbfVars[f"vbf_dR_j1_HVV"] = vbf2.deltaR(VVJet)
+            vbfVars[f"vbf_dR_j0_Hbb"] = vbf1.deltaR(bbJet)
+            vbfVars[f"vbf_dR_j1_Hbb"] = vbf2.deltaR(bbJet)
+
+            vbfVars[f"vbf_dR_jj"] = vbf1.deltaR(vbf2)
+            # print(f"\nTime aftter first using vbf1: {time.time()-start_time:.6f} seconds")
+            # vbfVars[f"vbf_dR_HVV_j0"] = VVJet.deltaR(vbf1)  # probably not necessary since we can just reverse order
+            # vbfVars[f"vbf_dR_HVV_j1"] = VVJet.deltaR(vbf2)
+            # vbfVars[f"vbf_dR_Hbb_j0"] = bbJet.deltaR(vbf1)
+            # vbfVars[f"vbf_dR_Hbb_j1"] = bbJet.deltaR(vbf2)
+
+            # print(f"\nTime taken computing jj stuff: {time.time()-start_time:.6f} seconds")
+            jj = vbf1 + vbf2
+            vbfVars[f"vbf_Mass_jj"] = jj.M
+            print(jj.M[0:10], "testing masses to see if we implemented this correctly.")
+            vbfVars[f"vbf_dEta_jj"] = np.abs(vbf1.eta - vbf2.eta)
+
+            # print(f"\nTime taken inserting several: {time.time()-start_time:.6f} seconds")
+
+            # trying to compute j1_CMF stuff manually since system_4vec is very slow to use.
```

Comment:
I'm surprised by this - you're saying using the internal `vector` or `coffea` methods are slower than doing this manually?

Reply:
At first, I had an issue where the vbf jet data was the incorrect dimension and this was causing the vector packages to do a lot of unnecessary computations. I made the alternative to try and fix this (without noticing the cause). My implementation was roughly the same speed. After I realized the bug and fixed it I tested it on a few files and my implementation took 0.5s per file vs 0.6s per file for vector so I kept it, but this might have just been a random fluctuation. 
I'm planning to change to the vector methods going forward since it is cleaner.